### PR TITLE
CaloTowerCalib: Flag Negative Energy Towers

### PR DESCRIPTION
**Pull request by Steepspace**
## Types of changes
[comment]: <> ( What types of changes does your code introduce? Put an `x` in all the boxes that apply: )
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work for users)
- [ ] Requiring change in macros repository (Please provide links to the macros pull request in the last section)
- [x] I am a member of [GitHub organization of sPHENIX Collaboration](https://github.com/orgs/sPHENIX-Collaboration/people), EIC, or ECCE (contact Chris Pinkenburg to join)

## What kind of change does this PR introduce? (Bug fix, feature, ...)

Flag towers below negative energy threshold as `badChi2`. Towers with severe negative reconstructed energy (e.g., from uncalibrated channels, pedestal fluctuations, or waveform fitting artifacts) can distort downstream clustering and jet reconstruction. This adds a negative energy threshold check to flag such towers with `isBadChi2`. Key changes:
- Flag towers where calibrated energy is below a configurable threshold as badChi2 (`set_isBadChi2(true)`).
- By default, the threshold is enabled and set to **-2.0 GeV**.
- Apply threshold checking exclusively to EMCal and HCal detectors (CEMC, HCALIN, HCALOUT).
- Add setter/getter methods to CaloTowerCalib:
    - `set_doNegEnergyThreshold(bool do_threshold)`: Allows to enable / disable negative energy threshold.
    - `set_negEnergyThreshold(float threshold)`: Set the threshold directly.
    - `get_negEnergyThreshold()` and `get_doNegEnergyThreshold()` getters.

## TODOs (if applicable)

[comment]: <> ( In case this is a draft PR, e.g. for running checks using Jenkins, please make the pull request as a draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/  )

## Motivation

Flag EMCal and HCal towers with calibrated energy below a configurable negative threshold. The default threshold is `-2.0 GeV`. This identifies problematic towers through `badChi2`.

## Key changes

- Added methods to enable or disable negative-energy thresholding.
- Added methods to set and retrieve the threshold.
- Removed the ambiguous setter overload.
- Applied the check to CEMC, HCALIN, and HCALOUT towers.
- Marked towers as bad with `set_isBadChi2()` when calibrated energy is below the threshold.
- Reused the calibrated tower pointer during event processing.

## Potential risk areas

- The default setting changes reconstruction behavior for towers below `-2.0 GeV`.
- No I/O format changes are indicated.
- The implementation adds one per-tower comparison. No material performance impact is expected.
- No new thread-safety concerns are apparent.

## Possible future improvements

- Add tests for configuration, detector selection, and threshold boundaries.
- Monitor the number of towers marked with `badChi2`.
- Document the threshold choice and its physics impact.

AI-generated summaries can contain mistakes. Contributors should verify the implementation and physics behavior against the source code and collaboration conventions.

### CODE REVIEW

**Comment by Steepspace** [comment]
@coderabbitai review

File: offline/packages/CaloReco/CaloTowerCalib.cc [hunk 1]
```diff
@@ -272,28 +277,36 @@ int CaloTowerCalib::process_event(PHCompositeNode *topNode)
...
     {
-      _calib_towers->get_tower_at_channel(channel)->set_energy(raw_amplitude * calibconst);
+      calib_energy = raw_amplitude * calibconst;
     }
-   
+    calib_tower->set_energy(calib_energy);
+
+    if (m_apply_neg_energy_threshold && calib_energy < m_negEnergyThreshold)
+    {
+      calib_tower->set_isBadChi2(true);
```
[the remark this answers is not in the document]
Comment (Steepspace) [reply offline/packages/CaloReco/CaloTowerCalib.cc:304]: All raw tower nodes (`TOWERS_CEMC`, `TOWERS_HCALIN`, `TOWERS_HCALOUT`) on DSTs are built using `TowerInfoContainerv4` (or `v2`), both of which implement status bits via `TowerInfov4` / `TowerInfov2`. `CaloTowerCalib` intentionally clones the input container (`_raw_towers->CloneMe()`) to preserve the container version.

**Comment by Steepspace** [comment]
@coderabbitai review

**Comment by Steepspace** [comment]
@coderabbitai review
